tests/sqlalchemy_dialect_compliance/test_dialect_compliance.py

```diff
@@ -42,8 +46,434 @@
+                ).where(
+                    uuid_table.c.uuid_text_data == data,
+                    uuid_table.c.uuid_text_data_nonnative == data,
+                )
+            ).first()
+            eq_((row[0].lower(), row[1].lower()), (data, data))
+
+        def test_literal_uuid(self, literal_round_trip):
+            data = str(uuid.uuid4())
+            literal_round_trip(String(), [data], [data])
+
+        def test_literal_text(self, literal_round_trip):
+            data = str(uuid.uuid4())
+            literal_round_trip(
+                String(),
+                [data],
+                [data],
+                filter_=lambda x: x.lower(),
+            )
+
+        def test_literal_nonnative_uuid(self, literal_round_trip):
+            data = str(uuid.uuid4())
+            literal_round_trip(String(), [data], [data])
+
+        def test_literal_nonnative_text(self, literal_round_trip):
+            data = str(uuid.uuid4())
+            literal_round_trip(
+                String(),
+                [data],
+                [data],
+                filter_=lambda x: x.lower(),
+            )
+
+        @testing.requires.insert_returning
+        def test_uuid_returning(self, connection):
+            data = str(uuid.uuid4())
+            str_data = str(data)
+            uuid_table = self.tables.uuid_table
+
+            result = connection.execute(
+                uuid_table.insert().returning(
+                    uuid_table.c.uuid_data,
+                    uuid_table.c.uuid_text_data,
+                    uuid_table.c.uuid_data_nonnative,
+                    uuid_table.c.uuid_text_data_nonnative,
+                ),
+                {
+                    "id": 1,
+                    "uuid_data": data,
+                    "uuid_text_data": str_data,
+                    "uuid_data_nonnative": data,
+                    "uuid_text_data_nonnative": str_data,
+                },
+            )
+            row = result.first()
+
+            eq_(row, (data, str_data, data, str_data))
+
+    class StringTest(_StringTest):
+        def test_dont_truncate_rightside(
+            self, metadata, connection, expr=None, expected=None
+        ):
+            t = Table(
+                "t",
+                metadata,
+                Column("x", String(2)),
+                Column("id", Integer, primary_key=True),
+            )
+            t.create(connection)
+            connection.connection.commit()
+            connection.execute(
+                t.insert(),
+                [{"x": "AB", "id": 1}, {"x": "BC", "id": 2}, {"x": "AC", "id": 3}],
+            )
+            combinations = [("%B%", ["AB", "BC"]), ("A%C", ["AC"]), ("A%C%Z", [])]
+
+            for args in combinations:
+                eq_(
+                    connection.scalars(select(t.c.x).where(t.c.x.like(args[0]))).all(),
+                    args[1],
+                )
+
+    # from else statement ....
```

Comment:
NIT: 
It looks like this is some extraneous deletes that are tied to one of the else blocks. I wonder if there is a way to reorganize the content so that the comment is not necessary. This is a nit, so don't spend a lot of time on it if it is too complicated.